hsc/crawler.py

```diff
@@ -50,38 +89,34 @@ def get_submission_url(self, challenge_slug, submission_id):
 		return self.challenge_url.format(challenge_slug, submission_id)
 
 	def store_submission(self, file_name, code):
-		print(file_name)
 		os.makedirs(os.path.dirname(file_name), exist_ok=True)
 		with open(file_name, 'w') as text_file:
-			print(code, file=text_file)
-
-	def update_readme(self, challenge_name, readme_file_path, challenge_slug, file_name, file_extension):
-		problem_url = self.problem_url.format(challenge_slug)
-		text = self.problem_readme_text.format(challenge_name, problem_url, file_name, file_extension)
-		with open(readme_file_path, 'a') as text_file:
-			print(text, file=text_file)
-		with open(readme_file_path, 'r') as text_file:
+			text_file.write(code)
+
+	def update_readme(self, readme_file_path, problem_readme_text):
+		with open(readme_file_path, 'r+') as text_file:
 			lines = text_file.readlines()
+			lines.append(problem_readme_text)
 			sortedlines = lines[:4] + sorted(lines[4:])
```

Comment:
I am unable to recall the reason why did we choose 4? Why aren't we sorting the first 4 lines?

Reply:
heading of readme, a newline, heading of table and table formatter

Reply:
It looks like some kind of a hack. Can we think of some alternate solution for this where we need not keep track of the number of starting lines?

Reply:
We can count newlines in [`new _readme_text`](https://github.com/Nullifiers/Hackerrank-Solution-Crawler/blob/add_file_extensions/hsc/crawler.py#L13)

Reply:
That also looks like a hack. I was thinking of storing actual lines somewhere else from where we can extract the number of solutions already present, and then add new solutions and sort them, and then append it to with actual headers. But this seems like overkill. So if nothing good comes up right now, we can leave this as it is.

Reply:
Well, counting lines is better than hardcoding, so I already did it. In case you come up with anything better.. let me know.